lib/Parser/Parser.cpp

```diff
@@ -163,13 +168,75 @@ Token Lexer::getNextToken(std::string &ErrStr) {
     }
   }
 
-  if ((*Begin >= 'a' && *Begin <= 'z') || (*Begin == '.')) {
+  if ((*Begin >= 'a' && *Begin <= 'z') || (*Begin == '.') ||
+      (*Begin >= 'A' && *Begin <= 'Z')) {
     const char *TokenBegin = Begin;
     do {
       ++Begin;
     } while (Begin != End && ((*Begin >= 'a' && *Begin <= 'z') ||
-             (*Begin == '.')));
-    return Token{Token::Ident, TokenBegin, size_t(Begin - TokenBegin), APInt()};
+             (*Begin == '.') || (*Begin >= 'A' && *Begin <= 'Z')));
+    std::string DataFlowFact = StringRef(TokenBegin, Begin - TokenBegin);
+    const char *PatternBegin;
+    Token T;
```

Comment:
ok this code is better now, but don't declare a Token here and fill it in later, rather use the in-place constructor notation everywhere, as in return Token{.....}